src/interpreter/index.ts

```diff
@@ -446,19 +448,25 @@ export class Interpreter implements Expr.Visitor<BrsType>, Stmt.Visitor<BrsType>
         let requiredType = typeDesignators[name.charAt(name.length - 1)];
 
         if (requiredType && requiredType !== value.kind) {
-            return this.addError(
-                new TypeMismatch({
-                    message: `Attempting to assign incorrect value to statically-typed variable '${name}'`,
-                    left: {
-                        type: requiredType,
-                        location: statement.name.location,
-                    },
-                    right: {
-                        type: value,
-                        location: statement.value.location,
-                    },
-                })
-            );
+            if (requiredType === ValueKind.Int64 && value.kind === ValueKind.Int32) {
```

Comment:
I suspect we'll need to add an extra branch in `Callable#getSignatureMismatches()` to cover a function that accepts a 64-bit integer and receives a 32-bit:

```brightscript
function addOne(a as LongInteger) as LongInteger
    return a + 1
end function

three% = 3
addOne(three%)
```

I'm not sure how RBI handles the other cases, so please check them all:

| Declared Argument Type (`as ___`) | Passed Argument Type | Converts? |
| ----- | ----- | ----- |
| integer | integer | no |
| integer | float | |
| integer | double | |
| integer | longinteger | |
| float | integer | |
| float | float | no |
| float | double | |
| float | longinteger | |
| double | integer | |
| double | float | yes |
| double | double | no |
| double | longinteger | |
| longinteger | integer | yes |
| longinteger | float | |
| longinteger | double | |
| longinteger | longinteger | no |

Reply:
I've made changes and now it works as described below:

Declared Argument Type (as ___) | Passed Argument Type | Converts?
-- | -- | --
integer | integer | no
integer | float |  yes
integer | double |  fails
integer | longinteger |  fails
float | integer | yes
float | float | no
float | double |  fails
float | longinteger |  fails
double | integer |  fails
double | float | yes
double | double | no
double | longinteger |  fails
longinteger | integer | yes
longinteger | float |  fails
longinteger | double |  fails
longinteger | longinteger | no

On Roku we have next:
<img width="447" alt="Screenshot 2021-01-13 at 16 26 37" src="https://user-images.githubusercontent.com/26671643/104464952-2bae7c00-55bc-11eb-819c-a5bb6e01fc9f.png">


As you can see, on Roku we have a bit weird casting, so probably it would be difficult to do the same here. @sjbarag should we do the same in `BRS` interpreter?

Reply:
Yes!  `brs` should match the behavior seen in the reference implementation found on Roku devices as much as possible 😃

Reply:
Created a separate ticket (#609), as was discussed in private.